src/shim/jsonConfig.ts

```diff
@@ -119,8 +119,39 @@ export class ShimJsonConfig implements IJsonConfig {
         }
         try {
             const jsonConfig: IJsonConfig = JSON.parse(jsonString);
-            this.connectionString = jsonConfig.connectionString;
-            this.enableAutoCollectExceptions = jsonConfig.enableAutoCollectExceptions;
+            if (jsonConfig.connectionString !== undefined) {
```

Comment:
Why you check for undefined?, when we merge undefined should be discarded

Reply:
Checking for undefined here on values that can be configured by environment variable earlier in the file. I found in testing that they were being overridden without the undefined checks.